src/common/ManagedCommon/WindowHelpers.cs

```diff
@@ -35,5 +35,17 @@ public static void BringToForeground(IntPtr handle)
                 }
             }
         }
+
+        /// <summary>
+        /// Workaround for a WinUI bug on Windows 10 in which a window's top border is always
+        /// black. Calls <c>DwmExtendFrameIntoClientArea()</c> with a <c>cyTopHeight</c> of 2 to force
+        /// the window's top border to be visible.<br/><br/>
+        /// Has no visible effect on Windows 11.
+        /// </summary>
+        public static void ForceTopBorder1PixelInset(IntPtr handle)
+        {
+            var margins = new NativeMethods.MARGINS { cxLeftWidth = 0, cxRightWidth = 0, cyBottomHeight = 0, cyTopHeight = 2 };
```

Comment:
As a safety measure, I think we should still only run this on Windows 10. Perhaps a check to a check like the  "IsWindows11" function and renaming this function to ForceTopBorder1PixelInsetOnWindows10. Thoughts?

Reply:
I'm in favor, but since OSVersionHelper is down in Common.UI, it'd mean:

a) moving `WindowHelpers` down into `Common.UI`
b) moving `OSVersionHelper` up into `ManagedCommon`
c) Moving `ForceTopBorder1PixelInset()` out of `WindowHelpers` and into some new class in `Common.UI`.

I think I have a slight preference for b), because it makes sense to me to keep an OS version check in the top level helper library, instead of a UI-specific one (as Common.UI appears to be). I'm not super familiar with conventions in this repo though, so is there any particular preference?

Reply:
I think b) makes sense, since it doesn't make sense it's on a UI library. ManagedCommon makes more sense.